### PR TITLE
Support OSC owning tracking datas as well as other QoL improvements

### PR DESCRIPTION
**Pull request by yangwill**
Several significant changes to OSC
- OSC now owns tracking datas, this prevents segfaults if we want to use the OSC outside of a main executable (ie. make an OSC factory)
- OSC cost logging now uses the actual drake evaluators instead of manually recomputing the cost. This prevents accidental mismatches if we change the cost formulation
- Improved impact-invariant formulation
- Add OSC gains base class to contain the common cost matrices

- - -
This change is [Reviewable](https://reviewable.io/reviews/dairlab/dairlib/325)

### CODE REVIEW

File: systems/controllers/osc/operational_space_control.h [hunk 1]
```diff
@@ -376,15 +419,12 @@ class OperationalSpaceControl : public drake::systems::LeafSystem<double> {
   mutable double prev_distinct_fsm_state_ = -1;
   drake::solvers::LinearEqualityConstraint* blend_constraint_;
   drake::solvers::VectorXDecisionVariable epsilon_blend_;
+  //  drake::solvers::BoundingBoxConstraint*
```
Comment (yminchen) [inline systems/controllers/osc/operational_space_control.h:422]: dead codes

**Review by yminchen — CHANGES_REQUESTED** [review]

**Review by Brian-Acosta — CHANGES_REQUESTED** [review]
> Reviewed 18 of 20 files at r1, 9 of 9 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/dairlab/dairlib/325)* status: 27 of 29 files reviewed, 11 unresolved discussions (waiting on @yangwill and @yminchen)

___
*[`common/blending_utils.h` line 7 at r2](https://reviewable.io/reviews/dairlab/dairlib/325#-NGrof6V9uG-9EjzoX_6:-NGrof6V9uG-9EjzoX_7:br9hr9y) ([raw file](https://github.com/dairlab/dairlib/blob/9e3e932339de8b56e10676f62a5ce79955dec262/common/blending_utils.h#L7)):*
> ```objc
> double blend_sigmoid(double t, double tau, double window);
> 
> double blend_exp(double t, double tau, double window);
> ```

nit - Add newline
___
*[`common/blending_utils.cc` line 11 at r2](https://reviewable.io/reviews/dairlab/dairlib/325#-NGroguy2aywxfHfS0-7:-NGroguy2aywxfHfS0-8:b-g8j00a) ([raw file](https://github.com/dairlab/dairlib/blob/9e3e932339de8b56e10676f62a5ce79955dec262/common/blending_utils.cc#L11)):*
> ```cpp
> double blend_exp(double t, double tau, double window) {
>   return 1 - exp(-(t + window) / tau);
> }
> ```

nit - Add Newline
___
*[`common/discrete_time_filter.h` line 5 at r2](https://reviewable.io/reviews/dairlab/dairlib/325#-NGrpBbQ9OUj4Y5HRFV1:-NGrpBbRBdFRg18ntaNj:bpj4oqd) ([raw file](https://github.com/dairlab/dairlib/blob/9e3e932339de8b56e10676f62a5ce79955dec262/common/discrete_time_filter.h#L5)):*
> ```objc
> #include <Eigen/Core>
> 
> namespace dairlib {
> ```

Any reason this shouldn't go in systems/filters ?
___
*[`common/discrete_time_filter.cc` line 1 at r2](https://reviewable.io/reviews/dairlab/dairlib/325#-NGrpJQi5eBh1y4lwhfy:-NGrpJQi5eBh1y4lwhfz:b-vk5970) ([raw file](https://github.com/dairlab/dairlib/blob/9e3e932339de8b56e10676f62a5ce79955dec262/common/discrete_time_filter.cc#L1)):*
> ```cpp
> #include "discrete_time_filter.h"
> ```

see above about moving this file
___
*[`examples/Cassie/BUILD.bazel` line 370 at r1](https://reviewable.io/reviews/dairlab/dairlib/325#-NG30qBQz-0Qwudw1kOe-r1-370:-NGrpQjZ3IuTF6X1wnBK:b-ul2lov) ([raw file](https://github.com/dairlab/dairlib/blob/92c584b6ddc11627ca0327b8cf22d5728f2e7d27/examples/Cassie/BUILD.bazel#L370)):*
<details><summary><i>Previously, yminchen (Yu-Ming Chen) wrote…</i></summary><blockquote>

I think this would prevent CI from checking? If so, can we delete this line? 
</blockquote></details>

The idea here is to stop supporting/updating the walking controller since we only use the alip version at this point  
___
*[`examples/Cassie/run_osc_walking_controller_alip.cc` line 477 at r2](https://reviewable.io/reviews/dairlab/dairlib/325#-NGrpng203rsukFH0EVi:-NGrpng203rsukFH0EVj:b-stmt6p) ([raw file](https://github.com/dairlab/dairlib/blob/5eb23cf6138ac1a4b478178b0f544c01dea5b686/examples/Cassie/run_osc_walking_controller_alip.cc#L477)):*
> ```cpp
> 
>   auto vel_map = MakeNameToVelocitiesMap<double>(plant_w_spr);
>   swing_foot_data.AddJointAndStateToIgnoreInJacobian(
> ```

Thanks for doing this :)
___
*[`systems/controllers/osc/operational_space_control.h` line 141 at r2](https://reviewable.io/reviews/dairlab/dairlib/325#-NGrqmmY22i0RE9h0u4o:-NGrqmmY22i0RE9h0u4p:bbh5v50) ([raw file](https://github.com/dairlab/dairlib/blob/9e3e932339de8b56e10676f62a5ce79955dec262/systems/controllers/osc/operational_space_control.h#L141)):*
> ```objc
> 
>   // Regularization cost weights
>   void SetInputCostForJointAndFsmStateWeight(const std::string& joint_u_name, int fsm,
> ```

Probably enough regularization costs  by now that we should document what they all do

**Review by yangwill — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dairlab/dairlib/325)* status: 20 of 31 files reviewed, 9 unresolved discussions (waiting on @Brian-Acosta and @yminchen)

___
*[`common/discrete_time_filter.h` line 5 at r2](https://reviewable.io/reviews/dairlab/dairlib/325#-NGrpBbQ9OUj4Y5HRFV1:-NH6CbWr4KggPjmIhh4R:b4ghzro) ([raw file](https://github.com/dairlab/dairlib/blob/9e3e932339de8b56e10676f62a5ce79955dec262/common/discrete_time_filter.h#L5)):*
<details><summary><i>Previously, Brian-Acosta (Brian Acosta) wrote…</i></summary><blockquote>

Any reason this shouldn't go in systems/filters ?
</blockquote></details>

this isn't a leafsystem so i didn't put it in `systems`
___
*[`common/discrete_time_filter.cc` line 1 at r2](https://reviewable.io/reviews/dairlab/dairlib/325#-NGrpJQi5eBh1y4lwhfy:-NH6Ceng0rNDNjR723CM:b-896fix) ([raw file](https://github.com/dairlab/dairlib/blob/9e3e932339de8b56e10676f62a5ce79955dec262/common/discrete_time_filter.cc#L1)):*
<details><summary><i>Previously, Brian-Acosta (Brian Acosta) wrote…</i></summary><blockquote>

see above about moving this file
</blockquote></details>

Done.
___
*[`examples/Cassie/BUILD.bazel` line 370 at r1](https://reviewable.io/reviews/dairlab/dairlib/325#-NG30qBQz-0Qwudw1kOe-r1-370:-NH6Cg1YA79T2GMjwCmh:b-896fix) ([raw file](https://github.com/dairlab/dairlib/blob/92c584b6ddc11627ca0327b8cf22d5728f2e7d27/examples/Cassie/BUILD.bazel#L370)):*
<details><summary><i>Previously, Brian-Acosta (Brian Acosta) wrote…</i></summary><blockquote>

The idea here is to stop supporting/updating the walking controller since we only use the alip version at this point  
</blockquote></details>

Done.
___
*[`examples/Cassie/run_osc_walking_controller_alip.cc` line 477 at r2](https://reviewable.io/reviews/dairlab/dairlib/325#-NGrpng203rsukFH0EVi:-NH6JLb99StOZo7kUqgx:b-896fix) ([raw file](https://github.com/dairlab/dairlib/blob/5eb23cf6138ac1a4b478178b0f544c01dea5b686/examples/Cassie/run_osc_walking_controller_alip.cc#L477)):*
<details><summary><i>Previously, Brian-Acosta (Brian Acosta) wrote…</i></summary><blockquote>

Thanks for doing this :)
</blockquote></details>

Done.
___
*[`systems/controllers/osc/operational_space_control.h` line 397 at r1](https://reviewable.io/reviews/dairlab/dairlib/325#-NG30qBQz-0Qwudw1kOs-r1-397:-NH6EslAA317Vx41Iz-5:b-896fix) ([raw file](https://github.com/dairlab/dairlib/blob/92c584b6ddc11627ca0327b8cf22d5728f2e7d27/systems/controllers/osc/operational_space_control.h#L397)):*
<details><summary><i>Previously, yminchen (Yu-Ming Chen) wrote…</i></summary><blockquote>

There are more dead codes in this PR. Please remove them
</blockquote></details>

Done.
___
*[`systems/controllers/osc/operational_space_control.h` line 422 at r1](https://reviewable.io/reviews/dairlab/dairlib/325#-NG30qBQz-0Qwudw1kOs-r1-422:-NH6Es71DReBCUIOq2VA:b-896fix) ([raw file](https://github.com/dairlab/dairlib/blob/92c584b6ddc11627ca0327b8cf22d5728f2e7d27/systems/controllers/osc/operational_space_control.h#L422)):*
<details><summary><i>Previously, yminchen (Yu-Ming Chen) wrote…</i></summary><blockquote>

dead codes
</blockquote></details>

Done.
___
*[`systems/controllers/osc/operational_space_control.h` line 141 at r2](https://reviewable.io/reviews/dairlab/dairlib/325#-NGrqmmY22i0RE9h0u4o:-NH6JM1v12cyKrkDlsen:b-896fix) ([raw file](https://github.com/dairlab/dairlib/blob/9e3e932339de8b56e10676f62a5ce79955dec262/systems/controllers/osc/operational_space_control.h#L141)):*
<details><summary><i>Previously, Brian-Acosta (Brian Acosta) wrote…</i></summary><blockquote>

Probably enough regularization costs  by now that we should document what they all do
</blockquote></details>

Done.
___
*[`systems/controllers/osc/operational_space_control.cc` line 81 at r1](https://reviewable.io/reviews/dairlab/dairlib/325#-NG30qBQz-0Qwudw1kOr-r1-81:-NH6Ht_xCy7H1vS_SKqb:b-23qsvf) ([raw file](https://github.com/dairlab/dairlib/blob/92c584b6ddc11627ca0327b8cf22d5728f2e7d27/systems/controllers/osc/operational_space_control.cc#L81)):*
<details><summary><i>Previously, yminchen (Yu-Ming Chen) wrote…</i></summary><blockquote>

What does this clock do? Is it wall time or simulator context time? 
It might be good to add documentation explaining each port in the header file.

</blockquote></details>

Added a bunch of documentation
___
*[`systems/framework/impact_info_vector.h` line 16 at r1](https://reviewable.io/reviews/dairlab/dairlib/325#-NG30qBQz-0Qwudw1kOx-r1-16:-NH6J5KgDBn0wnEgAKfd:b-896fix) ([raw file](https://github.com/dairlab/dairlib/blob/92c584b6ddc11627ca0327b8cf22d5728f2e7d27/systems/framework/impact_info_vector.h#L16)):*
<details><summary><i>Previously, yminchen (Yu-Ming Chen) wrote…</i></summary><blockquote>

Need documentation for this class and its members.
</blockquote></details>

Done.

**Review by Brian-Acosta — APPROVED** [review]
small nit but otherwise 
:lgtm:

> Reviewed 11 of 11 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/dairlab/dairlib/325#-:-NH6T3W46HU6mq1mJHMA:b-tydrow)* status: all files reviewed, 10 unresolved discussions (waiting on @yangwill and @yminchen)

___
*[`systems/controllers/osc/operational_space_control.h` line 386 at r3](https://reviewable.io/reviews/dairlab/dairlib/325#-NH6SPvg8godXlT4rTrr:-NH6SPvhCDw_aG925S0y:b-hnmajp) ([raw file](https://github.com/dairlab/dairlib/blob/a97338975927fbb7bed2f5133735436245f5739a/systems/controllers/osc/operational_space_control.h#L386)):*
> ```objc
> 
>   // Solver
>   std::unordered_map<int, std::unique_ptr<drake::solvers::OsqpSolver>> solvers_;
> ```

Small nit, can we get rid of the map since we only use one solver?

**Review by yangwill — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dairlab/dairlib/325)* status: all files reviewed, 9 unresolved discussions (waiting on @Brian-Acosta and @yminchen)

___
*[`systems/controllers/osc/operational_space_control.h` line 386 at r3](https://reviewable.io/reviews/dairlab/dairlib/325#-NH6SPvg8godXlT4rTrr:-NH6atep2nG80SAYfaCF:b-3u0xzy) ([raw file](https://github.com/dairlab/dairlib/blob/a97338975927fbb7bed2f5133735436245f5739a/systems/controllers/osc/operational_space_control.h#L386)):*
<details><summary><i>Previously, Brian-Acosta (Brian Acosta) wrote…</i></summary><blockquote>

Small nit, can we get rid of the map since we only use one solver? 
</blockquote></details>

Ya oops, Done.

**Review by yminchen — CHANGES_REQUESTED** [review]
Thanks for making the changes! I added a few more comments and questions

> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/dairlab/dairlib/325#-:-NHW9ZKf4h9Hyl05ho4V:bb14vni)* status: all files reviewed, 13 unresolved discussions (waiting on @Brian-Acosta and @yangwill)

___
*[`examples/Cassie/BUILD.bazel` line 370 at r1](https://reviewable.io/reviews/dairlab/dairlib/325#-NG30qBQz-0Qwudw1kOe-r1-370:-NHW3RIZDEfPIawuC764:b-vhs0ux) ([raw file](https://github.com/dairlab/dairlib/blob/92c584b6ddc11627ca0327b8cf22d5728f2e7d27/examples/Cassie/BUILD.bazel#L370)):*
<details><summary><i>Previously, yangwill (William Yang) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

I think LIP controller still serves as a good baseline controller since it's common and we understand LIP much better than ALIP. I'm also using it to compare against my ROM MPC from time to time.
___
*[`systems/controllers/osc/operational_space_control.h` line 386 at r3](https://reviewable.io/reviews/dairlab/dairlib/325#-NH6SPvg8godXlT4rTrr:-NHW6U-v0x8g87sShcQ2:b-y8fdk4) ([raw file](https://github.com/dairlab/dairlib/blob/a97338975927fbb7bed2f5133735436245f5739a/systems/controllers/osc/operational_space_control.h#L386)):*
<details><summary><i>Previously, yangwill (William Yang) wrote…</i></summary><blockquote>

Ya oops, Done.
</blockquote></details>

Discussed in person. We should keep the fastosqp (without copying over Drake's sparse matrices implementation) until Drake's osqp can do warmstart and give the same level of solution quality as fastosqp
___
*[`systems/controllers/osc/operational_space_control.cc` line 794 at r4](https://reviewable.io/reviews/dairlab/dairlib/325#-NHW7P0IAcixSfX9X1Ly:-NHW7P0JBnQPXuCOWnwc:bkd1iel) ([raw file](https://github.com/dairlab/dairlib/blob/d9b653dfa9bd299f43ea8db3a5afe88e630ab783/systems/controllers/osc/operational_space_control.cc#L794)):*
> ```cpp
>     *epsilon_sol_ = result.GetSolution(epsilon_);
>     initial_guess_x_[0] = result.GetSolution();
>     initial_guess_y_[0] = result.get_solver_details<OsqpSolver>().y;
> ```

Reading through the code, it seems to me that the keys for these two maps only contain 0?
___
*[`systems/controllers/osc/operational_space_control.cc` line 797 at r4](https://reviewable.io/reviews/dairlab/dairlib/325#-NHW7h_t6_P8f7SwIrH6:-NHW7h_t6_P8f7SwIrH7:b6n6b4u) ([raw file](https://github.com/dairlab/dairlib/blob/d9b653dfa9bd299f43ea8db3a5afe88e630ab783/systems/controllers/osc/operational_space_control.cc#L797)):*
> ```cpp
>   } else {
>     //    *u_prev_ = VectorXd::Zero(n_u_);
>     *u_prev_ = 0.99 * *u_sol_ + VectorXd::Random(n_u_);
> ```

0.99 is probably a parameter specific to Cassie. Should we use `*u_prev_ = *u_sol_` by default?

I'm also curious did you see `*u_prev_ = 0.99 * *u_sol_ + VectorXd::Random(n_u_);` improve the running controller?
___
*[`systems/controllers/osc/operational_space_control.cc` line 825 at r4](https://reviewable.io/reviews/dairlab/dairlib/325#-NHW8bQx2XNdxCb-KXsS:-NHW8bQx2XNdxCb-KXsT:btfvlc4) ([raw file](https://github.com/dairlab/dairlib/blob/d9b653dfa9bd299f43ea8db3a5afe88e630ab783/systems/controllers/osc/operational_space_control.cc#L825)):*
> ```cpp
>   for (unsigned int i = 0; i < all_contacts_.size(); i++) {
>     if (next_contact_set.find(i) != next_contact_set.end()) {
>       //      std::cout << i << std::endl;
> ```

dead code
___
*[`systems/controllers/osc/operational_space_control.cc` line 949 at r4](https://reviewable.io/reviews/dairlab/dairlib/325#-NHW8lrE6kYPHSoT3ViW:-NHW8lrE6kYPHSoT3ViX:bdzrhgm) ([raw file](https://github.com/dairlab/dairlib/blob/d9b653dfa9bd299f43ea8db3a5afe88e630ab783/systems/controllers/osc/operational_space_control.cc#L949)):*
> ```cpp
>   //  if (joint_limit_cost_) {
>   //    joint_limit_cost_->Eval(*dv_sol_, &y_joint_limit_cost);
>   //  }
> ```

Why did we comment out the joint limit cost? Should we add it back?
___
*[`systems/controllers/osc/operational_space_control.cc` line 966 at r4](https://reviewable.io/reviews/dairlab/dairlib/325#-NHW8rYV7MlETjbzvYQd:-NHW8rYV7MlETjbzvYQe:btfvlc4) ([raw file](https://github.com/dairlab/dairlib/blob/d9b653dfa9bd299f43ea8db3a5afe88e630ab783/systems/controllers/osc/operational_space_control.cc#L966)):*
> ```cpp
>   output->regularization_cost_names.clear();
> 
>   //  output->regularization_costs.reserve(4);
> ```

dead code

**Review by yangwill — COMMENTED** [review]
addressed your comments!

> *[Reviewable](https://reviewable.io/reviews/dairlab/dairlib/325#-:-NJ64MeMDyyGhFV8p18v:b-se6cwe)* status: 0 of 45 files reviewed, 13 unresolved discussions (waiting on @Brian-Acosta and @yminchen)

___
*[`examples/Cassie/BUILD.bazel` line 370 at r1](https://reviewable.io/reviews/dairlab/dairlib/325#-NG30qBQz-0Qwudw1kOe-r1-370:-NJ5v98G6kzd0XimBzOm:b-y7v7mu) ([raw file](https://github.com/dairlab/dairlib/blob/92c584b6ddc11627ca0327b8cf22d5728f2e7d27/examples/Cassie/BUILD.bazel#L370)):*
<details><summary><i>Previously, yminchen (Yu-Ming Chen) wrote…</i></summary><blockquote>

I think LIP controller still serves as a good baseline controller since it's common and we understand LIP much better than ALIP. I'm also using it to compare against my ROM MPC from time to time.
</blockquote></details>

ok i'll add it back
___
*[`systems/controllers/osc/operational_space_control.h` line 386 at r3](https://reviewable.io/reviews/dairlab/dairlib/325#-NH6SPvg8godXlT4rTrr:-NJ5vEdN92-Px5v38Nwp:b-896fix) ([raw file](https://github.com/dairlab/dairlib/blob/a97338975927fbb7bed2f5133735436245f5739a/systems/controllers/osc/operational_space_control.h#L386)):*
<details><summary><i>Previously, yminchen (Yu-Ming Chen) wrote…</i></summary><blockquote>

Discussed in person. We should keep the fastosqp (without copying over Drake's sparse matrices implementation) until Drake's osqp can do warmstart and give the same level of solution quality as fastosqp
</blockquote></details>

Done.
___
*[`systems/controllers/osc/operational_space_control.cc` line 794 at r4](https://reviewable.io/reviews/dairlab/dairlib/325#-NHW7P0IAcixSfX9X1Ly:-NJ5vJsq70YHdbCKXW13:bs2ukaj) ([raw file](https://github.com/dairlab/dairlib/blob/d9b653dfa9bd299f43ea8db3a5afe88e630ab783/systems/controllers/osc/operational_space_control.cc#L794)):*
<details><summary><i>Previously, yminchen (Yu-Ming Chen) wrote…</i></summary><blockquote>

Reading through the code, it seems to me that the keys for these two maps only contain 0?
</blockquote></details>

after testing i don't think the per-mode solver and warm start is beneficial if we assume the solutions are mostly continuous, so i'll revert to using a single solver and guess
___
*[`systems/controllers/osc/operational_space_control.cc` line 797 at r4](https://reviewable.io/reviews/dairlab/dairlib/325#-NHW7h_t6_P8f7SwIrH6:-NJ5vbfS7QGcGz24N1A9:b6jmep0) ([raw file](https://github.com/dairlab/dairlib/blob/d9b653dfa9bd299f43ea8db3a5afe88e630ab783/systems/controllers/osc/operational_space_control.cc#L797)):*
<details><summary><i>Previously, yminchen (Yu-Ming Chen) wrote…</i></summary><blockquote>

0.99 is probably a parameter specific to Cassie. Should we use `*u_prev_ = *u_sol_` by default?

I'm also curious did you see `*u_prev_ = 0.99 * *u_sol_ + VectorXd::Random(n_u_);` improve the running controller?
</blockquote></details>

i should remove that, i didn't want to use a failed solve as a warm start
___
*[`systems/controllers/osc/operational_space_control.cc` line 825 at r4](https://reviewable.io/reviews/dairlab/dairlib/325#-NHW8bQx2XNdxCb-KXsS:-NJ5vvbs6AbDAOVlPf2O:b-896fix) ([raw file](https://github.com/dairlab/dairlib/blob/d9b653dfa9bd299f43ea8db3a5afe88e630ab783/systems/controllers/osc/operational_space_control.cc#L825)):*
<details><summary><i>Previously, yminchen (Yu-Ming Chen) wrote…</i></summary><blockquote>

dead code
</blockquote></details>

Done.
___
*[`systems/controllers/osc/operational_space_control.cc` line 949 at r4](https://reviewable.io/reviews/dairlab/dairlib/325#-NHW8lrE6kYPHSoT3ViW:-NJ5w0uj0XCzfXaSQ9J0:b-e6xoxl) ([raw file](https://github.com/dairlab/dairlib/blob/d9b653dfa9bd299f43ea8db3a5afe88e630ab783/systems/controllers/osc/operational_space_control.cc#L949)):*
<details><summary><i>Previously, yminchen (Yu-Ming Chen) wrote…</i></summary><blockquote>

Why did we comment out the joint limit cost? Should we add it back?
</blockquote></details>

i think we need a discussion on how to best implement the joint limit cost, i'll add a TODO to remind us
___
*[`systems/controllers/osc/operational_space_control.cc` line 966 at r4](https://reviewable.io/reviews/dairlab/dairlib/325#-NHW8rYV7MlETjbzvYQd:-NJ5wIiK9HYX0yHp5t7n:b-896fix) ([raw file](https://github.com/dairlab/dairlib/blob/d9b653dfa9bd299f43ea8db3a5afe88e630ab783/systems/controllers/osc/operational_space_control.cc#L966)):*
<details><summary><i>Previously, yminchen (Yu-Ming Chen) wrote…</i></summary><blockquote>

dead code
</blockquote></details>

Done.

**Review by yminchen — APPROVED** [review]
LGTM after addressing the new comments

> Reviewed 19 of 42 files at r8, 4 of 13 files at r9, 5 of 7 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/dairlab/dairlib/325#-:-NKr45TB7LwRTG8upFlP:b6ybolm)* status: 28 of 45 files reviewed, 11 unresolved discussions (waiting on @Brian-Acosta and @yangwill)

___
*[`common/discrete_time_filter.h` line 7 at r10](https://reviewable.io/reviews/dairlab/dairlib/325#-NKqvUOxEsC0kKosnrkx:-NKqvUOxEsC0kKosnrky:b-ccz1ss) ([raw file](https://github.com/dairlab/dairlib/blob/322fcc03b89f1f8ce3b813a48759dcba5634c46f/common/discrete_time_filter.h#L7)):*
> ```objc
> namespace dairlib {
> 
> class DiscreteTimeFilter {
> ```

This class can have a documentation to save user's time. E.g. if the user wants a cutoff frequency of 50Hz, how does this relate to `alpha`?
___
*[`examples/Cassie/cassie_utils.h` line 70 at r10](https://reviewable.io/reviews/dairlab/dairlib/325#-NKqy6qi8RppzsTekbyu:-NKqy6qjC5zE527ISdlC:bttv7z) ([raw file](https://github.com/dairlab/dairlib/blob/2962073d277555186d4f224c032a5b280942ab7f/examples/Cassie/cassie_utils.h#L70)):*
> ```objc
> /// @param add_loop_closure_springs Whether or not to add the loop closure
> ///     distance constraint via stiff springs. Default = true.
> /// @param add_loop_closure_springs Whether or not to add the reflected inertia
> ```

shouldn't this be `add_reflected_inertia`
___
*[`examples/Cassie/multibody_sim.cc` line 106 at r10](https://reviewable.io/reviews/dairlab/dairlib/325#-NKqyayM9XaouUQsFkC7:-NKqyayM9XaouUQsFkC8:b8rsglu) ([raw file](https://github.com/dairlab/dairlib/blob/2962073d277555186d4f224c032a5b280942ab7f/examples/Cassie/multibody_sim.cc#L106)):*
> ```cpp
> 
>   plant.set_penetration_allowance(FLAGS_penetration_allowance);
>   plant.set_stiction_tolerance(FLAGS_v_stiction);
> ```

does TAMSI solver use these two flags? If so, should we add them back?

**Review by yangwill — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dairlab/dairlib/325)* status: 25 of 45 files reviewed, 11 unresolved discussions (waiting on @Brian-Acosta, @yangwill, and @yminchen)

___
*[`examples/Cassie/cassie_utils.h` line 70 at r10](https://reviewable.io/reviews/dairlab/dairlib/325#-NKqy6qi8RppzsTekbyu:-NKsLN-_7lxC96n0TnVR:b-yjlobo) ([raw file](https://github.com/dairlab/dairlib/blob/2962073d277555186d4f224c032a5b280942ab7f/examples/Cassie/cassie_utils.h#L70)):*
<details><summary><i>Previously, yminchen (Yu-Ming Chen) wrote…</i></summary><blockquote>

shouldn't this be `add_reflected_inertia`
</blockquote></details>

Done. Added comment describing the parameter
___
*[`examples/Cassie/multibody_sim.cc` line 106 at r10](https://reviewable.io/reviews/dairlab/dairlib/325#-NKqyayM9XaouUQsFkC7:-NKsLXELDbmPQdx7kBae:bks7d2g) ([raw file](https://github.com/dairlab/dairlib/blob/2962073d277555186d4f224c032a5b280942ab7f/examples/Cassie/multibody_sim.cc#L106)):*
<details><summary><i>Previously, yminchen (Yu-Ming Chen) wrote…</i></summary><blockquote>

does TAMSI solver use these two flags? If so, should we add them back?   
</blockquote></details>

Those lines were a mistake when i was trying drastically different stiffness and damping values. Should not be included

**Review by yangwill — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dairlab/dairlib/325)* status: 25 of 45 files reviewed, 11 unresolved discussions (waiting on @Brian-Acosta and @yminchen)

___
*[`common/discrete_time_filter.h` line 7 at r10](https://reviewable.io/reviews/dairlab/dairlib/325#-NKqvUOxEsC0kKosnrkx:-NKsPYNH4LqAuutzoQpU:b-896fix) ([raw file](https://github.com/dairlab/dairlib/blob/322fcc03b89f1f8ce3b813a48759dcba5634c46f/common/discrete_time_filter.h#L7)):*
<details><summary><i>Previously, yminchen (Yu-Ming Chen) wrote…</i></summary><blockquote>

This class can have a documentation to save user's time. E.g. if the user wants a cutoff frequency of 50Hz, how does this relate to `alpha`?
</blockquote></details>

Done.

**Pull request merged**